### PR TITLE
RedstoneGatewayInteractionsLoader implementation

### PR DESCRIPTION
**Pull request by ppedziwiatr**

### CODE REVIEW

File: src/__tests__/regression/gateway-interactions.test.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  it('returns same interactions ids for RedstoneGatewayLoader and ArweaveGatewayInteractionsLoader', async () => {
+    const smartweave = SmartWeaveNodeFactory.memCachedBased(arweave)
+      .setCacheableInteractionsLoader(new RedstoneGatewayInteractionsLoader('https://d1o5nlqr4okus2.cloudfront.net'))
+      .build();
+    const smartweave2 = SmartWeaveNodeFactory.memCachedBased(arweave)
+      .setCacheableInteractionsLoader(new ArweaveGatewayInteractionsLoader(arweave))
+      .build();
+    const { cacheableContractInteractionsLoader } = smartweave;
+    const { cacheableContractInteractionsLoader: cacheableContractInteractionsLoader2 } = smartweave2;
+    const response = await cacheableContractInteractionsLoader.load(contractId, 0, 655393);
```
[the remark this answers is not in the document]
Comment (asiaziola) [reply src/__tests__/regression/gateway-interactions.test.ts:61]: yeah... was about to change this one ;>

**Comment by ppedziwiatr** [comment]
as discussed, the SDK caching (`CacheableContractsInteractionsLoader`) needs to be switched off for now, as it may introduce some issues when loading only confirmed transactions (eg. if we ask for confirmed transactions 'till block 100, the SDK will cache the result at 100 block height. If then ask for the confirmed transactions 'till block 110, the SDK will ask for transactions from 101 to 110. If in the meantime some transactions below height 100 will be confirmed by the gateway - the will be skipped).
A solution should be probably to cache at highest block height from the response, not the "to" blockheight requested by the SDK.

File: src/__tests__/regression/gateway-interactions.test.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+LoggerFactory.INST.logLevel('fatal');
+
+const testCases: string[] = JSON.parse(fs.readFileSync(path.join(__dirname, 'test-cases.json'), 'utf-8'));
+
+/**
+ * These integration tests should verify whether ArweaveGatewayInteractionsLoader and RedstoneGatewayInteractionsLoader
+ * return same results for given variables
+ */
+describe.each(testCases)('contractId $a', (contractTxId) => {
+  it('returns same amount of interactions for RedstoneGatewayInteractionsLoader and ArweaveGatewayInteractionsLoader', async () => {
```
[the remark this answers is not in the document]
Comment (asiaziola) [reply src/__tests__/regression/gateway-interactions.test.ts:39]: i left it as it is and added a test to verify amount of interactions loaded for different block heights, guess it's quite well tested now :)

**Comment by ppedziwiatr** [comment]
@asiaziola , please squash the commit and amend the commit message to "feat: InteractionsLoader implementation for redstone-sw-gateway #59"

File: src/__tests__/regression/gateway-interactions.test.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  ArweaveGatewayInteractionsLoader,
+  DefaultEvaluationOptions
+} from '@smartweave';
+import { GQLEdgeInterface } from '../../legacy/gqlResult';
+
+/* 
+TODO: two test cases have been removed from the list - gateway-interaction test is failing due to the different
+amount of interactions returned from Redstone gateway and Arweave GQL gateway
+should be fixed in https://github.com/redstone-finance/redstone-sw-gateway/issues/17, following cases should be
+git bthen added to ../test-cases/gateway-interactions.json
```
Comment (ppedziwiatr) [inline src/__tests__/regression/gateway-interactions.test.ts:16]: say hwat?

**Comment by ppedziwiatr** [comment]
🙈

**Pull request merged**